fast64_internal/oot/oot_actor.py

```diff
@@ -1,47 +1,393 @@
+		bpy.context.region.tag_redraw()
+		self.report({'INFO'}, "Selected: " + self.itemChest)
+		return {'FINISHED'}
+
+	def invoke(self, context, event):
+		context.window_manager.invoke_search_popup(self)
+		return {'RUNNING_MODAL'}
+
+class OOT_SearchNaviMsgIDEnumOperator(bpy.types.Operator):
+	bl_idname = "object.oot_search_navi_msg_id_enum_operator"
+	bl_label = "Select Message ID"
+	bl_property = "naviMsgID"
+	bl_options = {'REGISTER', 'UNDO'}
 
+	naviMsgID : bpy.props.EnumProperty(items = ootNaviMsgID, default = "0x00")
+	objName : bpy.props.StringProperty()
+
+	def execute(self, context):
+		bpy.data.objects[self.objName].fast64.oot.actor.naviMsgID = self.naviMsgID
 		bpy.context.region.tag_redraw()
-		self.report({'INFO'}, "Selected: " + self.actorID)
+		self.report({'INFO'}, "Selected: " + self.naviMsgID)
 		return {'FINISHED'}
 
 	def invoke(self, context, event):
 		context.window_manager.invoke_search_popup(self)
 		return {'RUNNING_MODAL'}
 
+class OOTActorProperties(bpy.types.PropertyGroup):
+	# Each prop has its 'custom' variant because of the get and set functions
+	# Actors/Entrance Actor
+	actorID : bpy.props.EnumProperty(name='Actor ID', items=ootEnumActorID)
+	actorIDCustom : bpy.props.StringProperty(name='Actor Key', default='ACTOR_CUSTOM')
+	actorKey : bpy.props.StringProperty(name='Actor Key', default='0x0000')
+	actorParam : bpy.props.StringProperty(name = 'Actor Parameter', default = '0x0000',
+		get=lambda self: getValues(self, self.actorID, 'actorParam', 'Params', None),
+		set=lambda self, value: setValues(self, value, 'Params', 'actorID'))
+	actorParamCustom : bpy.props.StringProperty(name = 'Actor Parameter', default = '0x0000')
+
+	# Rotations
+	rotOverride : bpy.props.BoolProperty(name = 'Rot Override', default = False)
+	rotOverrideX : bpy.props.StringProperty(name = 'Rot X', default = '0x0',
+		get=lambda self: getValues(self, self.actorID, None, 'XRot', None),
+		set=lambda self, value: setValues(self, value, 'XRot', 'actorID'))
+	rotOverrideY : bpy.props.StringProperty(name = 'Rot Y', default = '0x0',
+		get=lambda self: getValues(self, self.actorID, None, 'YRot', None),
+		set=lambda self, value: setValues(self, value, 'YRot', 'actorID'))
+	rotOverrideZ : bpy.props.StringProperty(name = 'Rot Z', default = '0x0',
+		get=lambda self: getValues(self, self.actorID, None, 'ZRot', None),
+		set=lambda self, value: setValues(self, value, 'ZRot', 'actorID'))
+	rotOverrideCustom : bpy.props.BoolProperty(name = 'Override Rotation', default = False)
+	rotOverrideXCustom : bpy.props.StringProperty(name = 'Rot X', default = '0x0')
+	rotOverrideYCustom : bpy.props.StringProperty(name = 'Rot Y', default = '0x0')
+	rotOverrideZCustom : bpy.props.StringProperty(name = 'Rot Z', default = '0x0')
+
+	# We have to use a bool to know what's needed to be exported
+	XRotBool : bpy.props.BoolProperty(default=False)
+	YRotBool : bpy.props.BoolProperty(default=False)
+	ZRotBool : bpy.props.BoolProperty(default=False)
```

Comment:
why are these named "Bool"? I'm not sure I understand what these or the `rotxbool` variables are for

Reply:
it's used to let the export know which rotation it should export, the blender's one or the parameter one, but I guess I could just check if the value is not 0

Reply:
well checking if it's 0 isn't perfect because you may want to have 0
couldn't this be determined by looking in the xml data if the actor stores any data in its rotations instead?